packages/@glimmer/compiler/lib/passes/1-normalization/keywords/utils/if-unless.ts

```diff
@@ -67,7 +67,7 @@ function assertIfUnlessInlineKeyword(type: string) {
             inverted ? 'false' : 'true'
           }, and 3. the value to return if the condition is ${
             inverted ? 'true' : 'false'
-          }. Received ${positional?.size ?? 0} parameters`,
+          }. Received ${positional.size} parameters`,
```

Comment:
There's quite a bit of code in the repo that checked for things that can't happen. Sometimes, this
is because it's handling code from untyped JS users. In other cases, like this one, it's purely
internal code interacting with other typed code, and the check is a holdover from a previous time
when the check was necessary.

The current linting setup warns about situations like this. In some situations, like this one, I removed the unnecessary
check. In other situations, I disabled the lint and identified the fact that the check is for JS callers.